src/rust/cryptography-x509-validation/src/policy/mod.rs

```diff
@@ -119,28 +178,103 @@ pub struct Policy<'a, B: CryptoOps> {
     /// The validation time. All certificates validated by this policy must
     /// be valid at this time.
     pub validation_time: asn1::DateTime,
+
+    /// An extended key usage that must appear in EEs validated by this policy.
+    pub extended_key_usage: ObjectIdentifier,
+
+    /// The set of permitted signature algorithms, identified by their
+    /// algorithm identifiers.
+    ///
+    /// If not `None`, all certificates validated by this policy MUST
+    /// have a signature algorithm in this set.
+    ///
+    /// If `None`, all signature algorithms are permitted.
+    pub permitted_algorithms: Option<HashSet<AlgorithmIdentifier<'a>>>,
+
+    pub(crate) critical_ca_extensions: HashSet<ObjectIdentifier>,
+    pub(crate) critical_ee_extensions: HashSet<ObjectIdentifier>,
 }
 
 impl<'a, B: CryptoOps> Policy<'a, B> {
-    /// Creates a new policy with the given `CryptoOps`, an optional subject,
-    /// and a validation time.
+    /// Create a new policy with defaults for the certificate profile defined in
+    /// the CA/B Forum's Basic Requirements.
     pub fn new(ops: B, subject: Option<Subject<'a>>, time: asn1::DateTime) -> Self {
         Self {
             _ops: ops,
+            max_chain_depth: 8,
             subject,
             validation_time: time,
+            extended_key_usage: EKU_SERVER_AUTH_OID.clone(),
+            permitted_algorithms: Some(
+                WEBPKI_PERMITTED_ALGORITHMS
+                    .clone()
+                    .into_iter()
+                    .cloned()
+                    .collect(),
+            ),
+            critical_ca_extensions: RFC5280_CRITICAL_CA_EXTENSIONS.iter().cloned().collect(),
+            critical_ee_extensions: RFC5280_CRITICAL_EE_EXTENSIONS.iter().cloned().collect(),
         }
     }
 }
 
+impl<'a, B: CryptoOps> Policy<'a, B> {
+    /// Inform this policy of an expected critical extension in CA certificates.
+    ///
+    /// This allows the policy to accept critical extensions that the baseline
+    /// profile does not cover. The user is responsible for separately validating
+    /// these extensions.
+    pub fn assert_critical_ca_extension(mut self, oid: ObjectIdentifier) -> Self {
+        self.critical_ca_extensions.insert(oid);
+        self
+    }
+
+    /// Inform this policy of an expected critical extension in EE certificates.
+    ///
+    /// This allows the policy to accept critical extensions that the baseline
+    /// profile does not cover. The user is responsible for separately validating
+    /// these extensions.
+    pub fn assert_critical_ee_extension(mut self, oid: ObjectIdentifier) -> Self {
+        self.critical_ee_extensions.insert(oid);
+        self
+    }
```

Comment:
NB: These were included as intended APIs for the Rust side only; I can remove them to keep the diff small for the MVP if you'd prefer.

Reply:
Removed.